apps/spruce/src/hooks/useDimensions/index.ts

```diff
@@ -11,14 +11,17 @@ interface DimensionState {
  * @param ref The ref of the element to track the dimensions of.
  * @returns An object containing the current height and width of the ref.
  */
-export const useDimensions = (ref: MutableRefObject<HTMLElement>) => {
+export const useDimensions = <T extends HTMLElement>(
+  ref: MutableRefObject<T> | React.RefObject<T>,
+) => {
   const [state, setState] = useState<DimensionState>({ width: 0, height: 0 });
 
   const observer = useMemo(
     () =>
       new ResizeObserver((entries) => {
         requestAnimationFrame(() => {
-          const { height, width } = entries[0]?.contentRect ?? {};
```

Comment:
removed because [`contentRect`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/contentRect) is a legacy property that may be deprecated in the future